usr/src/uts/common/fs/zfs/zrlock.c

```diff
@@ -21,6 +21,7 @@
 /*
  * Copyright (c) 2010, Oracle and/or its affiliates. All rights reserved.
  * Copyright (c) 2014, 2015 by Delphix. All rights reserved.
+ * Copyright 2016 The MathWorks, Inc. All rights reserved.
```

Comment:
Copyright is not for credit. It is a legal thing. You only have a copyright on the code if you wrote it. Your work can (and should!) be credited in the commit message.

I think the most appropriate way to handle this would be as follows, assuming you made small changes at most:

1) Commit the code with an author of the original person who wrote the code. Include any copyright lines they used (which may be for their employer, if they wrote it on work time):
git commit --author "Will Andrews <will@freebsd.org>"

2) In the commit message, include a Signed-off-by tag:

If you made no changes, just a plain tag:
Signed-off-by: Youzhong Yang <yyang@mathworks.com>

If you made changes, do this:
Signed-off-by: Youzhong Yang <yyang@mathworks.com>
[yyang@mathworks.com: describe your changes here.]

If you made non-trivial changes, then you do hold a copyright on that. So include the original copyright notices, plus add your own (if you want, not required). Leave yourself as the author of the commit, and note in the commit message that your work was based upon work from Will Andrews <will@freebsd.org>.